native_client/generate_trie.cpp

```diff
@@ -8,7 +8,10 @@
 using namespace std;
 
 int generate_trie(const char* alphabet_path, const char* kenlm_path, const char* trie_path) {
-  Alphabet alphabet(alphabet_path);
+  Alphabet alphabet;
+  if (int err = alphabet.init(alphabet_path)) {
```

Comment:
feels like a death trap :-), maybe we should just:
```
int err = alphabet.init(alphabet_path);
if (err != 0) {
  return err;
}
```

Reply:
or just do the same than we do in `modelstate.cc`

Reply:
`modelstate.cc` does the same as is done here. By death trap do you mean the code style with the declaration inside the condition? I can change that.

Reply:
(Although I don't see the death trappiness of it)

Reply:
Yes, this is what I mean. Well, I always find that construction tricky, and error-prone in future evolutions.